### PR TITLE
Refactor OTXAnomaly

### PR DESCRIPTION
**Pull request by ashwinvaidya17**
### Summary

Makes `OTXModel` the base class of `OTXAnomaly`.

### Checklist

- [ ] I have added unit tests to cover my changes.​
- [ ] I have added integration tests to cover my changes.​
- [ ] I have ran e2e tests and there is no issues.
- [ ] I have added the description of my changes into CHANGELOG in my target branch (e.g., [CHANGELOG](https://github.com/openvinotoolkit/training_extensions/blob/develop/CHANGELOG.md) in develop).​
- [ ] I have updated the documentation in my target branch accordingly (e.g., [documentation](https://github.com/openvinotoolkit/training_extensions/tree/develop/docs) in develop).
- [ ] I have [linked related issues](https://help.github.com/en/github/managing-your-work-on-github/linking-a-pull-request-to-an-issue#linking-a-pull-request-to-an-issue-using-a-keyword).

### License

- [ ] I submit _my code changes_ under the same [Apache License](https://github.com/openvinotoolkit/training_extensions/blob/develop/LICENSE) that covers the project.
      Feel free to contact the maintainers if that's a concern.
- [ ] I have updated the license header for each file (see an example below).

```python
# Copyright (C) 2024 Intel Corporation
# SPDX-License-Identifier: Apache-2.0
```

### CODE REVIEW

**Review by sovrasov — APPROVED** [review]

File: src/otx/algo/anomaly/padim.py [hunk 1]
```diff
@@ -132,3 +131,16 @@ def predict_step(
         if not isinstance(inputs, dict):
             inputs = self._customize_inputs(inputs)
         return AnomalibPadim.predict_step(self, inputs, batch_idx, **kwargs)  # type: ignore[misc]
+
+    def forward(
```
Comment (eunwoosh) [inline src/otx/algo/anomaly/padim.py:135]: If this function is neccesary for both padim and stfpm, how about moving it to `OTXAnomaly` class?

File: src/otx/core/model/anomaly.py [hunk 2]
```diff
@@ -154,37 +155,6 @@ def configure_callbacks(self) -> list[Callback]:
             ),
         ]
 
-    def on_test_batch_end(
```
Comment (eunwoosh) [inline src/otx/core/model/anomaly.py:157]: It's just a question. Is it ok to remove? It seems that there is no corresponding code addition to this.

**Review by eunwoosh — COMMENTED** [review]
Thanks for your work :) I left some comments. Please take a look.

File: src/otx/algo/anomaly/padim.py (hunk 1 above)
Comment (ashwinvaidya17) [reply src/otx/algo/anomaly/padim.py:135]: The problem is that it calls `self.validation_step` which ends up calling OTXModel's validation step. We need AnomalibModule's validation step

File: src/otx/core/model/anomaly.py (hunk 2 above)
Comment (ashwinvaidya17) [reply src/otx/core/model/anomaly.py:157]: Yeah. I think we can remove configure callbacks as well. I'll update the PR

**Review by eunwoosh — APPROVED** [review]

File: src/otx/core/model/anomaly.py (hunk 2 above)
Comment (ashwinvaidya17) [reply src/otx/core/model/anomaly.py:157]: Yeah. I just tested it right now.

**Review by kprokofi — APPROVED** [review]

**Pull request merged**